### PR TITLE
Make sure test container is running before exec

### LINKED ISSUE #16598: Non-deterministic TestExecStartFails test
See [this log](https://jenkins.dockerproject.org/job/Docker-PRs/16646/console):

```
FAIL: docker_cli_exec_test.go:615: DockerSuite.TestExecStartFails

docker_cli_exec_test.go:622:
    c.Fatal("exec error message not received. The daemon might had crashed")
... Error: exec error message not received. The daemon might had crashed
```

### PR DESCRIPTION
**Pull request by cpuguy83**
Fixes race in TestExecStartFails

Fixes #16598

### CODE REVIEW

**Comment by vdemeester** [comment]
LGTM

File: integration-cli/docker_cli_exec_test.go [hunk 1]
```diff
@@ -616,9 +616,10 @@ func (s *DockerSuite) TestExecStartFails(c *check.C) {
 	testRequires(c, DaemonIsLinux)
 	name := "exec-15750"
 	dockerCmd(c, "run", "-d", "--name", name, "busybox", "top")
+	c.Assert(waitRun(name), check.IsNil)
 
-	_, errmsg, status := dockerCmdWithStdoutStderr(nil, "exec", name, "no-such-cmd")
-	if status == 255 && !strings.Contains(errmsg, "executable file not found") {
-		c.Fatal("exec error message not received. The daemon might had crashed")
+	out, status, _ := dockerCmdWithError("exec", name, "no-such-cmd")
+	if status == 255 && !strings.Contains(out, "executable file not found") {
```
Comment (duglin) [inline integration-cli/docker_cli_exec_test.go]: This check is a bit odd. If status != 255 (let's say its zero), the 'if' will fail and the test will pass.  Seems like it should be:
```
if status != 255 || !string.Contains(...) {
```
no?

Comment (cpuguy83) [inline integration-cli/docker_cli_exec_test.go]: Could probably drop the status all together, I think.

Comment (duglin) [inline integration-cli/docker_cli_exec_test.go]: can't hurt to make sure we get a non-zero rc tho.

**Comment by cpuguy83** [comment]
@duglin Fixed

**Comment by duglin** [comment]
LGTM

**Comment by duglin** [comment]
@vdemeester can you re-confirm your LGTM - then you can merge

**Comment by vdemeester** [comment]
@duglin confirming my LGTM, but still can't merge :sweat_smile: I still don't have the write access :stuck_out_tongue_closed_eyes:.

**Comment by duglin** [comment]
LOL they're just teasing you - I'll merge once janky is done

**Pull request merged**